### DISCUSSION TITLE
fatal: Authentication failed for ...

### DISCUSSION THREAD

**Discussion by wobcoug** [community/community#60927] [New to GitHub]
I am a first time user to GitHub, being introduced to it in a Codecademy course.  I followed the video explanations and everything worked until I tried to "git push -u origin main".  It took the user name, but when I entered the password, I got Authentication failed every time.  I tried changing the password, because my password had special characters.  I made sure I was typing it in correctly by logging out of GitHub and back in manually.  I even rebooted the computer to see if there was something hanging it up.

I am using the terminal in Visual Studio Code.

Thx.

**Comment by Gustavonn07** [comment]
If you're experiencing authentication issues when trying to push to GitHub via the terminal in Visual Studio Code, there are a few steps you can take to troubleshoot the issue:

1. Check your credentials: Make sure you are entering the correct username and password. Remember that the password is case-sensitive, so make sure you're typing it correctly, including case-sensitive.
You can use this commands to add your credential:
`git config --global user.name yourName`
`git config --global user.email youremail@gmail.com`

2. Confirm your authentication: Make sure you have permission to push to the repository. You can try re-authenticating by typing the git push command and providing your credentials when prompted. This will allow you to securely enter the password without it appearing in the terminal.

3. Check GitHub security settings: In some cases, specific security settings in GitHub can interfere with command-line authentication. Make sure you've enabled authentication via HTTPS or SSH, depending on how you've set up your repository. Also make sure your two-factor authentication setup isn't interfering with the process.

4. Consider using token-based authentication: If you are still experiencing authentication issues, a workaround is to use token-based authentication instead of password. On GitHub, you can generate a personal access token and use that to authenticate instead of the password. To do this, go to your GitHub security settings, generate a personal access token with the proper permissions, and use it as your password when pushing.

5. Check your internet connection: Sometimes authentication issues can occur due to unstable internet connection. Make sure you are connected to the internet correctly and try again.

Hope it helps!!

**Comment by wobcoug** [comment]
Thanks a ton.  I was able to get it done.

BR

Rob

**Reply by Gustavonn07** [reply]
hope it all worked out! If so, you can mark it as the best answer so that other person with the same problem can see it and let them know that it has already been answered (and would really help me too)!
